src/api/movieService.ts

```diff
@@ -1,33 +1,45 @@
-import {movieList} from "../assets/data/constData";
-import {Movie} from "../models/movie";
+import { movieList } from "../assets/data/constData";
+import { Movie } from "../models/movie";
 
+let movies = [...movieList];
 
 export const fetchMovies: () => Promise<Movie[]> = () => {
-  return new Promise((res,rej) => {
-    return setTimeout(() => res([...movieList]), 0);
+  return new Promise((res, rej) => {
+    return setTimeout(() => res(movies), 0);
   });
 };
 
 export const addMovie = (newMovie: Movie) => {
-  return new Promise((res, rej) => setTimeout(() => {
-    movieList.push(newMovie);
-    return res(true);
-  }, 1000))
+  return new Promise((res, rej) =>
+    setTimeout(() => {
+      let updatedList = [...movies];
+      updatedList.push(newMovie);
+      movies = updatedList;
+      return res(true);
+    }, 1000)
+  );
 };
 
 export const editMovie = (movie: Movie, id: string | undefined) => {
```

Comment:
```suggestion
export const editMovie = (movie: Movie, id: string) => {
```
Can `id` really be not defined?